Kernel/Term.cpp

```diff
@@ -866,26 +870,31 @@ Term* Term::create(unsigned function, unsigned arity, const TermList* args)
 {
   ASS_EQ(env.signature->functionArity(function), arity);
 
-  Term* s = new(arity) Term;
-  s->makeSymbol(function,arity);
-
-  bool share = true;
-  TermList* ss = s->args();
+  bool share = range(0, arity).all([&](auto i) { return argSafeToShare(args[i]); });
 
-  const TermList* curArg = args;
-  const TermList* argStopper = args+arity;
-  while (curArg!=argStopper) {
-    *ss = *curArg;
-    --ss;
-    if (!curArg->isSafe()) {
-      share = false;
+  auto allocTerm = [&]() {
```

Comment:
Do you, guys, know whether capturing word-sized creatures (ints, pointers) in lambdas by reference makes a difference, performance-wise to passing-by value?

Reply:
i don't know for sure but i would be surprised if the compiler creates an actual reference in a case like this. Here the lambda should be completely inlined as it's just called in the `Set::rawFindOrInsert`. But in general (say if the closure was stored somewhere and the reference is only used later) it might make a difference i reckon. But that's all just what I think should be going on so not 100% sure.

Reply:
Agree with Joe, resolving.

Reply:
Yes, sure. Here I actually just wanted to learn what you think :)

Reply:
Ah - you were perfectly clear, I was too terse in my previous comment! Aside: something that bit me before is capturing variables by *dangling* reference is particularly easy - be careful if you decide to capture-by-reference.